xarray/backends/h5netcdf_.py

```diff
@@ -175,10 +175,12 @@ def prepare_variable(self, name, variable, check_encoding=False,
                                           raise_on_invalid=check_encoding)
         kwargs = {}
 
-        if encoding.get('zlib'):
-            encoding['compression'] = 'gzip'
-            del encoding['zlib']
-            encoding['compression_opts'] = encoding.pop('complevel', None)
+        # Convert from NetCDF4-Python style compression settings to h5py style
+        if encoding.pop('zlib', False):
+            # If both styles are used together, h5py takes precedence
```

Comment:
@shoyer I changed the code, but I don't know how to write a test for it?

Reply:
@shoyer ping - please advise how I should write a unit test for ``check_encoding=True`` - is it something that can be requested from ``DataArray.to_netcdf``?

Reply:
Yes, `check_encoding=True` is triggered when `encoding` is supplied via `to_netcdf()` rather than on variables directly.

For example, this should be an error: `ds.to_netcdf(..., encoding={'foo': {'zlib': True', 'compression': 'other'}})`

Reply:
done